clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/ValueCodecFactory.java

```diff
@@ -79,10 +88,14 @@ public T decode(Object input) {
       ValueWrapper data = (ValueWrapper) input;
       ByteArrayInputStream bais = new ByteArrayInputStream(data.getValue());
       try {
-        try (ObjectInputStream ois = new ObjectInputStream(bais)) {
+        try (ObjectInputStream ois =
+               (whiteListedClasses != null) ? new WhiteListObjectInputStream(bais, whiteListedClasses)
+                                                                  : new ObjectInputStream(bais)) {
           @SuppressWarnings("unchecked")
           T result = (T) ois.readObject();
           return result;
+        } catch (InvalidClassException e) {
+          throw new RuntimeException(e.getMessage());
```

Comment:
Why are we obscuring the stacktrace?  Better to just wrap the exception.

Reply:
done